docs/tutorials/generate-custom-test-data.rst

```diff
@@ -0,0 +1,120 @@
+
+#. Use the ``curl`` command to post the ``impressions-key`` schema to the |sr|.
+
+   .. code:: bash
+
+      curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+      --data '{"schema": "{\"namespace\":\"streams\",\"name\":\"impressions\",\"type\":\"string\"}"}' \
+      http://localhost:8081/subjects/impressions-key/versions
+
+   Your output should resemble:
+
+   .. code:: bash
+
+      {"id":1}
+
+# Use the ``curl`` command to post the ``impressions-value`` schema to the |sr|.
+
+   .. code:: bash
+
+      curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+      --data '{"schema": "{\"namespace\":\"streams\",\"name\":\"impressions\",\"type\":\"record\",\"fields\":[{\"name\":\"impresssiontime\",\"type\":{\"type\":\"long\",\"format_as_time\":\"unix_long\",\"arg.properties\":{\"iteration\":{\"start\":1,\"step\":10}}}},{\"name\":\"impressionid\",\"type\":{\"type\":\"string\",\"arg.properties\":{\"regex\":\"impression_[1-9][0-9][0-9]\"}}},{\"name\":\"userid\",\"type\":{\"type\":\"string\",\"arg.properties\":{\"regex\":\"user_[1-9][0-9]?\"}}},{\"name\":\"adid\",\"type\":{\"type\":\"string\",\"arg.properties\":{\"regex\":\"ad_[1-9][0-9]?\"}}}]}"}' \
+      http://localhost:8081/subjects/impressions-value/versions
+
+   The return value is the identifier for the  Your output should resemble:
+
+   .. code:: bash
+
+      {"id":1}
+
+# Query the |sr| for the list of registered schemas:
+
+   .. code:: bash
+
+      curl -X GET http://localhost:8081/subjects
+
+   Your output should resemble:
+
+   .. code:: bash
+
+      ["impressions-value","impressions-key"]
+
+For more |sr| commands, see :ref:`_schemaregistry_intro`.
+
+Generate test data
+------------------
+
+When you have a custom schema registered, you can generate test data that's
+made up of random values that satisfy the schema requirements. In the
+``impressions`` schema, ad identifiers are two-digit random numbers between
+10 and 99, as specified by the regular expression ``ad_[1-9][0-9]``.
+
+Open a new command shell, and in the conlfluent/bin directory, start
+generating test values by using the ``ksql-datagen`` command. In this example,
+the schema file, impressions.avro, is in the same folder as ksql-datagen. 
+
+.. code:: bash
+
+    ./ksql-datagen schema=impressions.avro format=delimited topic=impressions key=impressionid propertiesFile=../etc/ksql/datagen.properties
+
+After a few startup messages, your output should resemble:
+
+.. code:: bash
+
+    impression_796 --> ([ 1528756317023 | 'impression_796' | 'user_41' | 'ad_29' ])
+    impression_341 --> ([ 1528756317446 | 'impression_341' | 'user_34' | 'ad_32' ])
+    impression_419 --> ([ 1528756317869 | 'impression_419' | 'user_58' | 'ad_74' ])
+    impression_399 --> ([ 1528756318146 | 'impression_399' | 'user_32' | 'ad_78' ])
+
+Consume the test data stream
+----------------------------
+
+In the KSQL query editor, create the ``impressions`` stream: ::

```

Comment:
these double colons are showing in the output: 

![image](https://user-images.githubusercontent.com/11722533/41300577-f36f70a0-6e1a-11e8-8153-d2299c7b48c3.png)

i'm not too familiar with using these (i usually just use `.. code::`). because you have it indented, it is working as a code block anyway.